### PR TITLE
DR 2915 puppeteer script for facelift homepage featured image

### PR DESCRIPTION
**Pull request by croyfish**
## Ticket:

- JIRA ticket [Create featured image test and have it run with the swimlanes test](https://jira.nypl.org/browse/DR-2915)

## This PR does the following:

- Adds a test script to assert that the featured image loads on the homepage

## How has this been tested?

- Do `npm install`
- Make sure you are accessinated and can reach the facelift qa environment
- Run the test against qa with `npm run homepageFeaturedImageLoadingTestQa`. If it does not time out, that means it succeeded. It should print all images loaded and the time taken, and it should succeed as long as the featured image loads and can be found in the DOM. This simulates a single user loading the homepage one time.
- Run the test against production with `npm run homepageFeaturedImageLoadingTestProd`. The above still applies.

## Accessibility concerns or updates

- None

### Checklist:

- [ x ] All new and existing tests passed.

### CODE REVIEW

File: test/homepageFeaturedImageLoadingTest.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  );
+
+  const loadTime = new Date() - startTime;
+  console.log(
+    "Time taken for all desired images to appear:",
+    loadTime,
+    "milliseconds"
+  );
+
+  console.log("Loaded image URLs:", loadedImageUrls);
```
Comment (7emansell) [inline test/homepageFeaturedImageLoadingTest.js:70]: if loadedImageUrls is empty should it still be printing the "Time taken for all desired images to appear" ?

Comment (croyfish) [reply test/homepageFeaturedImageLoadingTest.js:70]: Great question! I think if https://github.com/NYPL/digital-collections/blob/e1e5d1e6cbee3f36dd8f49d33c7690a89af038ef/test/homepageFeaturedImageLoadingTest.js#L52 times out (after 30s), the script exits with 1 and we don't even get that far?

Comment (7emansell) [reply test/homepageFeaturedImageLoadingTest.js:70]: i'm getting this, but can access qa-new-digitalcollections.nypl.org just fine? Is there something else I might be missing, it works fine with --prod Screenshot 2024-04-10 at 1 32 42 PM

Comment (croyfish) [reply test/homepageFeaturedImageLoadingTest.js:70]: Thanks, I can take a look

Comment (croyfish) [reply test/homepageFeaturedImageLoadingTest.js:70]: Hmm, I ran it just now and it worked fine. I am on the VPN and accessinated to nypl-dams-dev. Are you at the library today?

Comment (croyfish) [reply test/homepageFeaturedImageLoadingTest.js:70]: If you are, can you try clearing your cache and loading the qa page to see if the images are there?

Comment (7emansell) [reply test/homepageFeaturedImageLoadingTest.js:70]: yes I am at the library. On qa page swimlanes images load right away, featured image loads slowly

Comment (7emansell) [reply test/homepageFeaturedImageLoadingTest.js:70]: But definitely within 5-10s, not 30s

Comment (croyfish) [reply test/homepageFeaturedImageLoadingTest.js:70]: I can take another look at this. In any event, you shouldn't be getting an empty list of Loaded image URLs. Can you please add your public IP on that network to the ticket linked in the PR writeup?

Comment (7emansell) [reply test/homepageFeaturedImageLoadingTest.js:70]: Actually it looks like it only happens on ethernet, works fine on the other office networks

**Review by 7emansell — APPROVED** [review]
Weird behavior with on prem ethernet/qa environment to be looked into, but works great otherwise

File: test/homepageFeaturedImageLoadingTest.js (hunk 1 above)
Comment (croyfish) [reply test/homepageFeaturedImageLoadingTest.js:70]: Spawned: https://jira.nypl.org/browse/DR-2917

**Pull request merged**